modules/drip.cpp

```diff
@@ -0,0 +1,211 @@
+#include "drip.h"
+#include <math.h>
+
+using namespace daisysp;
+
+#define WUTR_SOUND_DECAY 0.95
+#define WUTR_SYSTEM_DECAY 0.996
+#define WUTR_GAIN 1.0
+#define WUTR_NUM_SOURCES 10.0
+#define WUTR_CENTER_FREQ0 450.0
+#define WUTR_CENTER_FREQ1 600.0
+#define WUTR_CENTER_FREQ2 750.0
+#define WUTR_RESON 0.9985
+#define WUTR_FREQ_SWEEP 1.0001
+#define MAX_SHAKE 2000
+
+int Drip::my_random(int max)
+{
+    return (rand() % (max + 1));
+}
+
+float Drip::noise_tick()
+{
+    float temp;
+    temp = 1.0 * rand() - 1073741823.5;
+    return temp * (1.0 / 1073741823.0);
+}
+
+void Drip::Init(float sample_rate, float dettack)
+{
+    sample_rate_ = sample_rate;
+    float temp;
+    dettack_   = dettack;
+    num_tubes_ = 10;
+    damp_      = 0.2;
+    shake_max_ = 0;
+    freq_      = 450.0;
+    freq1_     = 600.0;
+    freq2_     = 720.0;
+    amp_       = 0.3;
+
+    snd_level_   = 0.0;
+    float tpidsr = 2.0 * M_PI / sample_rate_;
+
+    kloop_     = (sample_rate_ * dettack_);
+    outputs00_ = 0.0;
+    outputs01_ = 0.0;
+    outputs10_ = 0.0;
+    outputs11_ = 0.0;
+    outputs20_ = 0.0;
+    outputs21_ = 0.0;
+
+    total_energy_ = 0.0;
+
+    center_freqs0_ = res_freq0_ = WUTR_CENTER_FREQ0;
+    center_freqs1_ = res_freq1_ = WUTR_CENTER_FREQ1;
+    center_freqs2_ = res_freq2_ = WUTR_CENTER_FREQ2;
+    num_objects_save_ = num_objects_ = WUTR_NUM_SOURCES;
+    sound_decay_                     = WUTR_SOUND_DECAY;
+    system_decay_                    = WUTR_SYSTEM_DECAY;
+    temp    = logf(WUTR_NUM_SOURCES) * WUTR_GAIN / WUTR_NUM_SOURCES;
+    gains0_ = gains1_ = gains2_ = temp;
+    coeffs01_                   = WUTR_RESON * WUTR_RESON;
+    coeffs00_ = -WUTR_RESON * 2.0 * cos(WUTR_CENTER_FREQ0 * tpidsr);
+    coeffs11_ = WUTR_RESON * WUTR_RESON;
+    coeffs10_ = -WUTR_RESON * 2.0 * cos(WUTR_CENTER_FREQ1 * tpidsr);
+    coeffs21_ = WUTR_RESON * WUTR_RESON;
+    coeffs20_ = -WUTR_RESON * 2.0 * cos(WUTR_CENTER_FREQ2 * tpidsr);
+
+    shake_energy_ = amp_ * 1.0 * MAX_SHAKE * 0.1;
+    shake_damp_   = 0.0;
+    if(shake_energy_ > MAX_SHAKE)
+        shake_energy_ = MAX_SHAKE;
+    shake_max_save_ = 0.0;
+    num_objects_    = 10;
+    finalZ0_ = finalZ1_ = finalZ2_ = 0.0;
+}
+
+float Drip::Process(bool trig)
```

Comment:
Normal double->float cleanup.

Use  `PI_F` from dsp.h instead of `M_PI`

Change `cos` usage to `cosf`

And add `f` to float literals